doc/constraint_causal_discovery.rst

```diff
@@ -0,0 +1,136 @@
+.. _constraint_causal_discovery:
+
+==================================
+Constraint-based causal discovery
+==================================
+.. automodule:: dodiscover.constraint
+    :no-members:
+    :no-inherited-members:
+.. currentmodule:: dodiscover
+
+The following are a set of methods intended for (non-parametric) structure learning
+of causal graphs (i.e. causal discovery) given observational and/or interventional data
+by checking constraints in the form of conditional independences (CI). At a high level
+all constraint-based causal discovery, tests CI statements, where we use different CI
+statistical tests to test the following null hypothesis:
+
+:math:`H_0: X \perp Y | Z` and :math:`H_A: X \not\perp Y | Z`
+
+For a given node ``X`` and ``Y`` in the underlying causal graph of interest, and
+a conditioning set, ``Z``. CI tests can have a variety of different assumptions
+that make one better than another in different settings and data assumptions.
+For more information on the CI tests themselves, see :ref:`conditional_independence`.
+
+Fundamental Assumptions of Constraint-Based Causal Discovery
+------------------------------------------------------------
+The fundamental assumptions of all algorithms in this section are the Markov property assumption
+and the causal faithfulness assumption :footcite:`Spirtes1993` and :footcite:`Pearl_causality_2009`.
+
+The Markov assumption states that all d-separation statements in the causal graph imply a
+corresponding CI statement in the data. This is a core-assumption that users from graphical modeling
```

Comment:
This is rather the faithfulness assumption and not the Markov assumption?

Reply:
Isn't it

Markov assumption = d-separation -> CI
whereas faithfulness = CI -> d-separation?

Reply:
Oh I see the confusion. I will reword this section.